### PR TITLE
Issue #1870: Use RequireJS to load the pedigree files

### PR DESCRIPTION
**Pull request by mjshepherd**

### CODE REVIEW

**Comment by antonkatz** [comment]
Who's nominated?

**Comment by mjshepherd** [comment]
hmmmm well since you were wondering @antonkatz @allasm

**Comment by DeanWay** [comment]
Build fails:
```
[INFO]   Verifying [PhenoTips/GroupsGadget.xml]... errors
[INFO]   - Parent must not be empty
```

**Comment by antonkatz** [comment]
The diff is too big...

**Comment by antonkatz** [comment]
+0.3 I just can't imagine being able to spot anything.

**Comment by marta-** [comment]
@antonkatz Did you try turning turning the white diff off with `?w=1` (e.g. s://github.com/mjshepherd/phenotips/commit/10cb1eadb21f3d6e239e4e9d8b0a593cbe10a4fe?w=1) ?

**Comment by antonkatz** [comment]
@marta- Nope. This is much better now.

File: components/pedigree/resources/src/main/resources/pedigree/filesaver/FileSaver.js [hunk 1]
```diff
@@ -13,264 +13,275 @@
...
-            saveTxtWindow = window.frames.saveTxtWindow;
-            if (!saveTxtWindow) {
-                saveTxtWindow = window.open('', '_temp', 'width=100,height=100');
-                if (!saveTxtWindow) {
-                    window.alert('Sorry, download file could not be created.');
-                    return false;
-                }
-            }
-        }
+  var FileSaver = {};
```
Comment (antonkatz) [inline components/pedigree/resources/src/main/resources/pedigree/filesaver/FileSaver.js]: Where is this object coming from? I could not see it in the master.

Comment (mjshepherd) [inline components/pedigree/resources/src/main/resources/pedigree/filesaver/FileSaver.js]: This is a new object that is exported at the end of this module. Originally this file just defined two global functions, now it exports those functions in this object.

File: components/pedigree/resources/src/main/resources/pedigree/view/exportSelector.js [hunk 2]
```diff
@@ -3,165 +3,173 @@
...
             }
-            var mimeType = "text/plain";
-        }
 
-        console.log("Export data: >>" + exportString + "<<");
-        if (exportString != "") {
-            saveTextAs(exportString, fileName);
+            console.log("Export data: >>" + exportString + "<<");
+            if (exportString != "") {
+                FileSaver.saveTextAs(exportString, fileName);
```
Comment (antonkatz) [inline components/pedigree/resources/src/main/resources/pedigree/view/exportSelector.js]: Was ```saveAs``` ever used?

Comment (antonkatz) [inline components/pedigree/resources/src/main/resources/pedigree/view/exportSelector.js]: Just wondering where it went. I think @allasm should be able to explain.

Comment (allasm) [inline components/pedigree/resources/src/main/resources/pedigree/view/exportSelector.js]: ```saveAs``` is used internally by ```saveTextAs```, which is used in exportSelector.js

File: components/pedigree/resources/src/main/resources/pedigree/pedigree.js [hunk 3]
```diff
@@ -6,914 +6,869 @@
...
+                    'tip' : 'Reason',
+                    'function' : 'setChildlessReason'
+                },
+                {
+                    'name' : 'consangr',
+                    'label' : 'Consanguinity of this relationship',
+                    'type' : 'radio',
+                    'values' : [
+                        { 'actual' : 'A', 'displayed' : 'Automatic' },
+                        { 'actual' : 'Y',
```
Comment (sdumitriu) [inline components/pedigree/resources/src/main/resources/pedigree/pedigree.js]: This one is lost.

**Pull request merged**